### PR TITLE
Docs: add operations/caching page

### PR DESCRIPTION
**Pull request by kvrhdn**
**What this PR does**:
This adds a _Caching_ section under _Deployment and Operations_. Intention is to gather recommendations and troubleshooting info about running proper caching with Tempo.

I guess this is also kind of a runbook? So maybe we should tackle this differently... I'm open for suggestions!

### CODE REVIEW

File: docs/tempo/website/operations/caching.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+---
+title: Caching
+weight: 6
+---
+
+# Caching
+
+Tempo can use an external cache to improve query performance.
+The supported implementations are [Memcached](https://memcached.org/) and [Redis](https://redis.io/). 
+Caching is used by the queriers and compactors.
```
Comment (achatterjee-grafana) [inline docs/tempo/website/operations/caching.md:10]:
```suggestion
Caching is used by the queriers and compactors.
Tempo can use an external cache to improve query performance.
The supported implementations are [Memcached](https://memcached.org/) and [Redis](https://redis.io/). 
```

File: docs/tempo/website/operations/caching.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+### Memcached
```
Comment (achatterjee-grafana) [inline docs/tempo/website/operations/caching.md:12]: Add information to this section.

File: docs/tempo/website/operations/caching.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+#### Connection limit
+
+As a cluster grows in size, the amount of instances of Tempo connecting to the cache servers will also increase.
+By default Memcached has a connection limit of 1024 and if this limit is exceeded new connections will be refused (resulting in HTTP 500 errors).
```
Comment (joe-elliott) [inline docs/tempo/website/operations/caching.md:17]: resulting in Tempo HTTP 500 errors?  Is there a log line or metric we could share here that would help people know when they're overwhelming memcached?

File: docs/tempo/website/operations/caching.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+This is resolved by increasing the connection limit of Memcached.
```
Comment (achatterjee-grafana) [inline docs/tempo/website/operations/caching.md:18]:
```suggestion
As a cluster grows in size, the number of instances of Tempo connecting to the cache servers also increases.
By default, Memcached has a connection limit of 1024. If this limit is surpassed new connections are refused (resulting in HTTP 500 errors).
This is resolved by increasing the connection limit of Memcached.
```

**Review by achatterjee-grafana — CHANGES_REQUESTED** [review]
Added doc change request.

File: docs/tempo/website/operations/caching.md (hunk 3 above)
Comment (kvrhdn) [reply docs/tempo/website/operations/caching.md]: Agree, we can share way more information to help out folks :+1:

**Comment by kvrhdn** [comment]
Thanks for the reviews! I've added a bit more detail.

I'm still not sure about the location of this section, it feels a bit like a troubleshooting page? It's very specific to caching though, so I don't want to drop it under troubleshooting/.

WDYT?

File: docs/tempo/website/operations/caching.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
+---
+title: Caching
+weight: 6
+---
+
+# Caching
+
+Caching is used by the queriers and compactors.
+Tempo can use an external cache to improve query performance.
```
Comment (annanay25) [inline docs/tempo/website/operations/caching.md:9]:
```suggestion
Caching is mainly used to improve query performance by storing bloom filters of all backend blocks which are accessed on every query.

Tempo uses an external cache to improve query performance.
```

**Comment by annanay25** [comment]
> I'm still not sure about the location of this section, it feels a bit like a troubleshooting page? It's very specific to caching though, so I don't want to drop it under troubleshooting/.


I vote for its current location (under Operations).

File: docs/tempo/website/operations/caching.md (hunk 2 above)
Comment (kvrhdn) [reply docs/tempo/website/operations/caching.md:13]: I've added a small introduction text.

**Review by joe-elliott — APPROVED** [review]

**Pull request merged**